rhel/matcher.go

```diff
@@ -25,15 +30,55 @@ func (*Matcher) Filter(record *claircore.IndexRecord) bool {
 }
 
 // Query implements driver.Matcher.
-func (*Matcher) Query() []driver.MatchConstraint {
-	return []driver.MatchConstraint{
-		driver.PackageModule,
-		driver.RepositoryName,
+func (m *Matcher) Query() []driver.MatchConstraint {
+	mcs := []driver.MatchConstraint{driver.PackageModule}
+	if m.ignoreUnpatched {
+		mcs = append(mcs, driver.HasFixedInVersion)
 	}
+	return mcs
+}
+
+// IsCPESubstringMatch is a Red Hat specific hack that handles the "CPE patterns" in the VEX
+// data. For historical/unfathomable reasons, Red Hat
+// doesn't use the syntax defined in the Matching Expression spec.
+// For example, "cpe:/a:redhat:openshift:4" is expected to match "cpe:/a:redhat:openshift:4.13::el8".
+//
+// This is defined (citation needed) to be a substring match on the "pattern" and "target" CPEs.
+// Since we always normalize CPEs into v2.3 "Formatted String" form, we need to trim the
+// added "ANY" attributes from the pattern.
+//
+// TODO(crozzy) Remove once RH VEX data updates CPEs with standard matching expressions.
+func isCPESubstringMatch(recordCPE cpe.WFN, vulnCPE cpe.WFN) bool {
+	return strings.HasPrefix(recordCPE.String(), strings.TrimRight(vulnCPE.String(), ":*"))
```

Comment:
The `*` is a literal, when we retrieve the value from the DB it has been formatted (i.e. `cpe:/a:redhat:openshift:4:*:*:*:*:*:*`, this will just remove the `:*`s not any actual information.

Reply:
Incoming CPEs will always be in 2.3 Formatted String binding, so removing trailing `:*` characters turns it into the populated substring.